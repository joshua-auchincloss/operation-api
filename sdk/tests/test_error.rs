use operation_api_sdk::{Enum, Error, Struct, namespace};

include!("./shared.rs");

#[derive(Enum, serde::Serialize, serde::Deserialize)]
#[fields(version = 1)]
pub enum ErrorCode {
    Foo = 1,
    Baz = 2,
}

#[derive(Struct, serde::Serialize, serde::Deserialize)]
#[fields(version = 1)]
pub struct KnownError {
    code: ErrorCode,
    desc: String,
}

#[derive(Error, serde::Serialize, serde::Deserialize)]
#[fields(version = 1)]
pub enum OperationError {
    Unknown {
        /// some nested doc
        desc: String,
    },
    Known(KnownError),
}

namespace! {
    "abc.corp.test" {
        KnownError, ErrorCode, OperationError, OperationErrorUnknown
    }
}

#[test]
fn smoke_error_code() {
    smoke_basic::<ErrorCode, _>("../samples/test-error-code.toml", |ser| {
        operation_api_testing::assert_yaml_snapshot!(ser)
    })
}

#[test]
fn smoke_error_unknown() {
    smoke_basic::<OperationErrorUnknown, _>(
        "../samples/test-struct-operation-error-unknown.toml",
        |ser| operation_api_testing::assert_yaml_snapshot!(ser),
    )
}

#[test]
fn smoke_known_error() {
    smoke_basic::<KnownError, _>("../samples/test-struct-known-error.toml", |ser| {
        operation_api_testing::assert_yaml_snapshot!(ser)
    })
}

#[test]
fn smoke_basic_error() {
    smoke_basic::<OperationError, _>("../samples/test-operation-error.toml", |ser| {
        operation_api_testing::assert_yaml_snapshot!(ser)
    })
}
