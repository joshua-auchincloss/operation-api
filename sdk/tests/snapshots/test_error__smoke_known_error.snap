---
source: sdk/tests/test_error.rs
expression: ser
---
type: struct@v1
name: KnownError
namespace: abc.corp.test
description: ~
version: 1
fields:
  code:
    name: code
    namespace: abc.corp.test
    description: ~
    version: ~
    type:
      compound_type:
        kind: enum
        ref: ErrorCode
    optional: false
  desc:
    name: desc
    namespace: abc.corp.test
    description: ~
    version: ~
    type: string
    optional: false
